app/parsers/bulkrax/oai_adventist_qdc_parser.rb

```diff
@@ -72,31 +75,35 @@ def build_records
       @collections = []
       @works = []
       @file_sets = []
-
       if model_field_mappings.map { |mfm| records.first.metadata.find("//#{mfm}").first }.any?
-
         records.map do |r|
           model_field_mappings.each do |model_mapping|
             next unless r.metadata.find("//#{model_mapping}").first
 
-            if r.header.set_spec.present?
-              generate_collection r.header.set_spec
+            generate_collection r.header.set_spec if r.header.set_spec.present?
+            if r.metadata.find("//#{model_mapping}").first.content.casecmp('collection').zero?
+              @collections << r
             elsif r.metadata.find("//#{model_mapping}").first.content.casecmp('fileset').zero?
               @file_sets << r
             else
               @works << r
             end
           end
         end
-        @collections = @collections.flatten.compact.uniq
-        @file_sets = @file_sets.flatten.compact.uniq
-        @works = @works.flatten.compact.uniq
+        flatten_arrays(%w[@collections @file_sets @works])
       else # if no model is specified, assume all records are works
-        @works = records.flatten.compact.uniq
+        @works = records
```

Comment:
why do we not want works to equal `records.flatten.compact.uniq` as in the previous version of this line and in #164?

Reply:
On the following line, Braydon introduced a flatten arrays method call.  That handles the logic in a consistent manner as above on line 93.

Reply:
but `flatten_arrays` doesn't get called on `records`; that I can see anyway.

and since line 95 is inside of the `else` conditional, then line 93 won't have run to flatten the works either correct?